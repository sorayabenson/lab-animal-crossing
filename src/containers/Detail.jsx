import React, { Component } from 'react'
import VillagerDetail from '../components/villagers/VillagerDetail'
import { villagerTicket } from '../services/animalCrossingAPI';

export default class Detail extends Component {
    state = {
        loading: true,
        villager: {}
    }

    async componentDidMount() {
        const villager = await villagerTicket(this.props.match.params.id);

        this.setState({
            loading: false,
            villager: villager
        })
    }
    
    render() {
        const { loading, villager } = this.state;

        if (loading) return <h1>a visitor is arriving!</h1>

        return (
            <VillagerDetail villager={villager}/>
        )
    }
}
