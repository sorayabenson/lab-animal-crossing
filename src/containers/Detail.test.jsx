import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import Detail from './Detail';

describe('Detail render', () => {
  afterEach(() => cleanup());

  it('renders a villager', async () => {
    render(
      <Detail
        match={{
          params: {
            id: '5f5fb4bbbfd05c2aed82e526',
          },
        }}
      />
    );
    screen.getByText('a visitor is arriving!');
    const div = await screen.findAllByRole('group', { name: 'villager' });
    expect(div).toMatchSnapshot();
  });
});
