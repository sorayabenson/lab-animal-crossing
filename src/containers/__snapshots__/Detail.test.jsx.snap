// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail render renders a villager 1`] = `
Array [
  <div
    aria-label="villager"
    role="group"
  >
    <img
      alt="Kiki"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c5/Kiki_NH.png/revision/latest?cb=20200801231206"
    />
    <h3>
      Kiki
    </h3>
    <p>
      "
      Nothing ventured, nothing gained.
      "
    </p>
    <p>
      personality: 
      normal
    </p>
    <p>
      species: 
      cat
    </p>
    <p>
      skill: 
      Fishing
    </p>
    <a
      href="https://animalcrossing.fandom.com/wiki/Kiki"
    >
      Kiki
      's secret diary
    </a>
  </div>,
]
`;
